tests/test_views.py

```diff
@@ -0,0 +1,74 @@
+"""Tests for the views functions."""
+from datetime import datetime
+from unittest.mock import Mock, patch
+
+from django.contrib.auth import get_user_model
+from django.test import Client, TestCase
+from django.urls import reverse
+from django.utils import timezone
+
+from blog.models import Post
+
+
+class ViewsTest(TestCase):
+    """Tests for views in Blog application."""
+
+    def setUp(self):
+        """Initialize test data: client, user, posts with different dates."""
+        self.user = get_user_model().objects.create()
+        self.client = Client()
+        self.tm_zone = timezone.get_current_timezone()
+
+        self.past_post = Post.objects.create(
+            author=self.user,
+            title='test past post',
+            published_date=datetime(2016, 1, 4, tzinfo=self.tm_zone),
+        )
+
+        self.current_post = Post.objects.create(
+            author=self.user,
+            title='test current post',
+            published_date=datetime(2024, 3, 1, tzinfo=self.tm_zone),
+        )
+
+        self.future_post = Post.objects.create(
+            author=self.user,
+            title='test future post',
+            published_date=datetime(2030, 3, 1, tzinfo=self.tm_zone),
+        )
+
+    def test_post_list_rendering(self):
+        """Check correct posts displayed based on mocked current time."""
+        post_list_url = reverse('post_list')
+        posts_in_different_time = {
+            # before past post
+            datetime(2015, 1, 2, tzinfo=self.tm_zone):
+                [],
+            # after past post
+            datetime(2024, 2, 1, tzinfo=self.tm_zone):
+                [self.past_post],
+            # after current post
+            datetime(2024, 3, 2, tzinfo=self.tm_zone):
+                [self.past_post, self.current_post],
+            # in the future
+            datetime(2030, 4, 1, tzinfo=self.tm_zone):
+                [self.past_post, self.current_post, self.future_post],
+        }
+        for now, expected_posts in posts_in_different_time.items():
+            with self.subTest(msg=f'The post was published on {now!s}'):
```

Comment:
This description refers to an unidentified post but each loop iteration works with a list of posts. Which is it? I thought it's about time being frozen, not some post having a date.

Reply:
Edited.

Reply:
Marked.